components/script/dom/timeranges.rs

```diff
@@ -61,11 +61,13 @@ pub struct TimeRangesContainer {
     ranges: Vec<TimeRange>,
 }
 
-impl TimeRangesContainer {
-    pub fn new() -> Self {
+impl Default for TimeRangesContainer {
+    fn default() -> Self {
         Self { ranges: Vec::new() }
     }
+}
```

Comment:
You should be able to just add `Default` to the `#[derive(...)]` decorator of `TimeRangesContainer`.

Reply:
> #[derive(...)]

Hello @fabricedesre, thank you for looking at this.

Please correct me if I get it wrong, do you mean the decorator as in 
```
#[derive(Clone, Debug, JSTraceable, MallocSizeOf)]
pub struct TimeRangesContainer {
    ranges: Vec<TimeRange>,
}

impl Default for TimeRangesContainer {
    fn default() -> Self {
        Self { ranges: Vec::new() }
    }
}
```
can be updated to ```#[derive(Default, Clone, Debug, JSTraceable, MallocSizeOf)]``` and then one wouldn't have to replace every call to ``` TimeRangesContainer::new()``` with ```TimeRangesContainer::default()``` in the test files ?